src/main/java/org/wso2/okta/client/OktaOAuthClient.java

```diff
@@ -0,0 +1,1046 @@
+                reader = new BufferedReader(new InputStreamReader(entity.getContent(), OktaConstants.UTF_8));
+                responseJSON = getParsedObjectByReader(reader);
+
+                if (responseJSON == null) {
+                    log.error(String.format("Invalid token %s", accessToken));
+                    tokenInfo.setTokenValid(false);
+                    tokenInfo.setErrorcode(APIConstants.KeyValidationStatus.API_AUTH_INVALID_CREDENTIALS);
+                    return tokenInfo;
+                }
+                tokenInfo.setTokenValid((Boolean) responseJSON.get(OktaConstants.ACCESS_TOKEN_ACTIVE));
+
+                if (tokenInfo.isTokenValid()) {
+                    long expiryTime = (Long) responseJSON.get(OktaConstants.ACCESS_TOKEN_EXPIRY);
+                    long issuedTime = (Long) responseJSON.get(OktaConstants.ACCESS_TOKEN_ISSUED);
+                    tokenInfo.setValidityPeriod(expiryTime - issuedTime);
+
+                    String tokScopes = (String) responseJSON.get(OktaConstants.ACCESS_TOKEN_SCOPE);
+
+                    if (StringUtils.isNotEmpty(tokScopes)) {
+                        tokenInfo.setScope(tokScopes.split("\\s+"));
+                    }
+
+                    tokenInfo.setIssuedTime(issuedTime);
+                    tokenInfo.setConsumerKey((String) responseJSON.get(OktaConstants.CLIENT_ID));
+                    tokenInfo.setEndUserName((String) responseJSON.get(OktaConstants.ACCESS_TOKEN_USER_NAME));
+                    tokenInfo.addParameter(OktaConstants.ACCESS_TOKEN_SUBJECT,
+                            responseJSON.get(OktaConstants.ACCESS_TOKEN_SUBJECT));
+                    tokenInfo.addParameter(OktaConstants.ACCESS_TOKEN_AUDIENCE,
+                            responseJSON.get(OktaConstants.ACCESS_TOKEN_AUDIENCE));
+                    tokenInfo.addParameter(OktaConstants.ACCESS_TOKEN_ISSUER,
+                            responseJSON.get(OktaConstants.ACCESS_TOKEN_ISSUER));
+                    tokenInfo.addParameter(OktaConstants.ACCESS_TOKEN_TYPE,
+                            responseJSON.get(OktaConstants.ACCESS_TOKEN_TYPE));
+                    tokenInfo.addParameter(OktaConstants.ACCESS_TOKEN_USER_ID,
+                            responseJSON.get(OktaConstants.ACCESS_TOKEN_USER_ID));
+                    tokenInfo.addParameter(OktaConstants.ACCESS_TOKEN_IDENTIFIER,
+                            responseJSON.get(OktaConstants.ACCESS_TOKEN_IDENTIFIER));
+
+                    return tokenInfo;
+                }
+            } else {
+                log.error(String.format("Invalid token %s", accessToken));
+                tokenInfo.setTokenValid(false);
+                tokenInfo.setErrorcode(APIConstants.KeyValidationStatus.API_AUTH_INVALID_CREDENTIALS);
+                return tokenInfo;
+            }
+        } catch (ParseException e) {
+            handleException(OktaConstants.ERROR_WHILE_PARSE_RESPONSE + e.getMessage(), e);
```

Comment:
Why are we duplicating. e.getMessage() ? Whe you pass e, it will has those details as well. Please correct it in other places as well